elephant/phase_analysis.py

```diff
@@ -190,3 +191,178 @@ def spike_triggered_phase(hilbert_transform, spiketrains, interpolate):
+
+    Parameters:
+    -----------
+    phases_data: list of neo.AnalogSignals objects with multiple trials
+        time-series of two signals with n trials each
+        # version_1:
+        axis: 0 -> trial, 1 -> signal x/y, 2 -> phases
+
+    Returns:
+    --------
+    plv: array-like object
+        phase-locking value (float)
+        range: [0, 1]
+
+    Notes
+    -----
+    This implementation is based on the formula taken from [1] (pp. 195).
+
+    PLV_t = 1/N * abs(sum_n=1_to_N(exp{i * theta(t, n)} ) )
+
+    where theta(t, n) is the phase difference phi_x(t, n) - phi_y(t, n).
+
+    References:
+    -----------
+    [1] Jean-Philippe Lachaux, Eugenio Rodriguez, Jacques Martinerie,
+    and Francisco J. Varela, "Measuring Phase Synchrony in Brain Signals"
+    Human Brain Mapping, vol 8, pp. 194-208, 1999.
+    """
+    # version_1: phase_data has shape(trial, signal x & y, phases)
+    if not isinstance(phase_data[0][0], neo.AnalogSignal):
+        raise ValueError(
+            "structure of the data is not correct: 0-axis should be trials, "
+            "1-axis units and 2-axis neo AnalogSignals")
+    data_as_array = np.squeeze(np.asarray(phase_data))
+    for i in range(len(phase_data)):
+        if np.size(data_as_array[i], axis=0) != 2:
+            raise ValueError("trial number of signal x and y must be equal")
+    phases_x, phases_y = np.split(data_as_array, 2, axis=1)
+    if np.shape(phases_x) != np.shape(phases_y):
+        raise ValueError("trial length within signal x and y must be equal")
+
+    srate = phase_data[0][0].sampling_rate
+    plv = phase_locking_value(np.squeeze(phases_x), np.squeeze(phases_y))
+    plv_analog_signal = neo.AnalogSignal(signal=plv, units=pq.dimensionless,
+                                         sampling_rate=srate)
+    return plv_analog_signal
+
+
+def mean_phase_vector(phases, axis=0):
+    """
+    Calculates the mean vector of phases.
+
+    This function expects phases (in radians) and uses their representation as
+    complex numbers to calculate the direction 'theta' and the length 'r'
+    of the mean vector.
+
+    Parameters
+    ----------
+    phases: array-like object
+        phases in radians
+    axis: {0, 1, None}
+        axis along which the mean_vector will be calculated
+        - None: across flattened array
+        - 0: across columns of array (default)
+        - 1: across rows of array
+
+    Returns
+    -------
+    z_mean_theta: array-like object
+        angle of the mean vector
+        range: (-pi, pi]
+    z_mean_r: array-like object
+        length of the mean vector
+        range: [0, 1]
+    """
+    # use complex number representation
+    # z_phases = np.cos(phases) + 1j * np.sin(phases)
+    z_phases = np.exp(1j * np.asarray(phases))
+    z_mean = np.mean(z_phases, axis=axis)
+    z_mean_theta = np.angle(z_mean)
+    z_mean_r = np.abs(z_mean)
+    return z_mean_theta, z_mean_r
+
+
+def phase_difference(alpha, beta):
```

Comment:
We discussed to avoid using greek letters. Since this function takes a pair of phases, couldn't this just be `phases_i` and `phases_j`?

Reply:
yes, it should be phase_i and _j
I fixed this